controllers/space/space_controller.go

```diff
@@ -0,0 +1,343 @@
+func (r *Reconciler) addFinalizer(logger logr.Logger, space *toolchainv1alpha1.Space) error {
+	// Add the finalizer if it is not present
+	if !util.HasFinalizer(space, toolchainv1alpha1.FinalizerName) {
+		logger.Info("adding finalizer on Space")
+		util.AddFinalizer(space, toolchainv1alpha1.FinalizerName)
+		if err := r.Client.Update(context.TODO(), space); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// ensureNSTemplateSet creates the NSTemplateSet on the target member cluster if it does not exist,
+// and updates the space's status accordingly.
+// returns `true` after creating the NSTemplateSet and until it is `ready`
+func (r *Reconciler) ensureNSTemplateSet(logger logr.Logger, space *toolchainv1alpha1.Space) (bool, error) {
+	if space.Spec.TargetCluster == "" {
+		return false, fmt.Errorf("missing target member cluster for Space '%s'", space.Name)
+	}
+	memberCluster, found := r.MemberClusters[space.Spec.TargetCluster]
+	if !found {
+		return false, fmt.Errorf("unknown target member cluster '%s'", space.Spec.TargetCluster)
+	}
+	name := space.Name // NSTemplateSet will have the same name as the Space
+	logger = logger.WithValues("target_member_cluster", space.Spec.TargetCluster, "name", name)
+	// create if not found
+	nsTmplSet := &toolchainv1alpha1.NSTemplateSet{}
+	if err := memberCluster.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: memberCluster.OperatorNamespace,
+		Name:      space.Name,
+	}, nsTmplSet); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("creating NSTemplateSet on target member cluster")
+			if err := r.setStatusProvisioning(space); err != nil {
+				return false, err
+			}
+			if nsTmplSet, err = r.newNSTemplateSet(memberCluster.OperatorNamespace, space); err != nil {
+				return false, err
+			}
+
+			if err := memberCluster.Client.Create(context.TODO(), nsTmplSet); err != nil {
+				logger.Error(err, "failed to create NSTemplateSet on target member cluster")
+				return false, r.setStatusNSTemplateSetCreationFailed(space, err.Error())
+			}
+			logger.Info("NSTemplateSet created on target member cluster")
+			return true, nil
+		}
+		return false, r.wrapErrorWithStatusUpdate(logger, space, r.setStatusNSTemplateSetCreationFailed, err, "failed to get NSTemplateSet '%s'", name)
+	}
+	logger.Info("NSTemplateSet already exists")
+
+	readyCond, found := condition.FindConditionByType(nsTmplSet.Status.Conditions, toolchainv1alpha1.ConditionReady)
+	// if UserAccount ready condition is already set to false, then don't update if no message is provided
+	// also, don't update when NSTemplateSet ready condition is in an invalid state
+	if !found || readyCond.Status == corev1.ConditionUnknown || readyCond.Status == corev1.ConditionFalse {
+		logger.Info("NSTemplateSet is not ready", "ready-condition", readyCond)
+		if err := r.setStatusFromNSTemplateSet(space, readyCond.Reason, readyCond.Message); err != nil {
+			return false, err
+		}
+		return true, nil
```

Comment:
> // if UserAccount ready

It's apparent that it was partially copied from UserAccount controller, but not entirely. 
Anyway, if the condition is either not found or unknown, then we should not replicate it to Space CR.
In UserAccount controller, we don't replicate the status if the UserAccount ready condition is already set to false to just not overload the synchronization mechanism. We could technically do the same also here, but it's not super needed https://github.com/codeready-toolchain/member-operator/blob/master/controllers/useraccount/useraccount_controller.go#L404-L416

Reply:
I removed call to update status of Space resource in faedeeb since it's not necessary